### DISCUSSION TITLE
How to use swagger ui and api's

### DISCUSSION THREAD

**Discussion by ni3aswal** [ansibleguy76/ansibleforms#214] [Q&A] [answered]
on swagger ui , generate a token and then login on the same ui , or create a small function in js to post the job or get list of jobs. (all api's are failing)

working from ansible-forms only

**Comment by ansibleguy76** [comment]
You did read the manual ? => "Bearer " + token
![image](https://github.com/user-attachments/assets/b2eaec45-3b9d-423b-a43e-0a839dd0b51c)

**Reply by ansibleguy76** [reply]
anyway, if ansibleforms works, the api works, because the frontend uses only the backend-api.

**Reply by ansibleguy76** [reply, marked as answer]
First use the **/auth/login** api and authorize.  

![image](https://github.com/user-attachments/assets/6be88e03-a9b1-47b1-ba64-56bf66fcb55b)  
![image](https://github.com/user-attachments/assets/f1cc0f1f-d001-44ca-adc1-15e479158e41)  
  
Choose **Try out**   
And click **Execute**  
  
![image](https://github.com/user-attachments/assets/703fc98f-7681-4190-8ca0-a341dec8209b)  
  
Capture the token  
Open the authorization for the api you want to test  
Preceed the token with **Bearer**  
  
![image](https://github.com/user-attachments/assets/fb8f19c2-380c-414c-a5de-07fabce5ae4e)  
![image](https://github.com/user-attachments/assets/3a866f64-e868-4d3f-8d10-37edb15cc6c4)  
  
Again choose **try out** + **execute**  
  
![image](https://github.com/user-attachments/assets/33b2fa27-faed-458d-85b2-313bcb02ca06)

**Comment by ni3aswal** [comment]
Thanks this helps!

**Reply by ansibleguy76** [reply]
Can you tell me where the issue was, so I can spot it for others to follow ?

**Reply by ni3aswal** [reply]
> Can you tell me where the issue was, so I can spot it for others to follow ?

i was trying to generate a token without login using admin user first. 
steps i followed to make it work:
1. login using admin user 
2. generate token
3. use that token and logout from admin.
4. then all apis worked, even my app worked. 

thanks

**Reply by ansibleguy76** [reply]
If you trust your environment, you can create a token with high expirydate.  As admin you can create one in the swagger api.
If you want one as non-admin.  Start ansible forms with a high expiry date (Environment variables).  Request a token, and then reset the Environment variable back to normal.  This way, you token is longlived and no auth is required.  However : only do this if your trust the environment.  Once the token is comprimised, you can't undo it and you will need to change the secret.

![image](https://github.com/user-attachments/assets/999b9f1a-d323-4e8c-b50a-8360f0c8f341)

**Comment by slekkus75** [comment]
Trying to create a token with which to run jobs scheduled. But keep getting unauthorized, even using an admin token.
Steps I do:
- Authorize: login with admin/password
- Get the token
- Authorize with the bearer token
- Execute whoami

I did change the env vars to extend the expiration time.

![image](https://github.com/user-attachments/assets/4743a3fa-25a5-442b-a719-aff469332f78)

![image](https://github.com/user-attachments/assets/1ea318ed-44c2-490a-b334-ce96f2a7dc85)

![image](https://github.com/user-attachments/assets/e3673217-35d4-464d-b4e9-d744663d2f59)

**Reply by ansibleguy76** [reply]
and it only happens if you extend the expiration?   you did prepend the token with the word Bearer, as explained at the top of the swagger page?

**Reply by slekkus75** [reply]
> and it only happens if you extend the expiration? you did prepend the token with the word Bearer, as explained at the top of the swagger page?

Missed the part where you stated to prepend "Bearer". Now it works. 
Just a notice:  have to include __inventory__ in the call, while __ansibleCredentials__, __playbook__ and __Credentials__ are automagically added.

```
{
  "formName": "Update Linux Servers",
  "extravars": {
    "clienthostgroup": "updategroup", 
    "__inventory__":"linuxupdates/inventory/freeipa.py",
    "moduscheck": "true"
  }
}
```
